frontend/public/components/conditions.tsx

```diff
@@ -40,6 +42,6 @@ export const Conditions: React.SFC<ConditionsProps> = ({obj}) => {
 
 /* eslint-disable no-undef */
 export type ConditionsProps = {
-  obj: any,
+  obj: any[],
```

Comment:
I'd rather call this property `conditions` instead of `obj`. Might be better to call the component `ConditionsTable`.

Reply:
Not quite sure how to do that.  If I replace `obj` with `conditions` here and in the component header `export const Conditions: React.SFC<ConditionsProps> = ({conditions})...`, I get
```
ERROR in /home/dtaylor/repos/openshift/origin/src/github.com/openshift/console/frontend/public/components/routes.tsx(253,17):
TS2322: Type '{ obj: {}[]; }' is not assignable to type 'IntrinsicAttributes & ConditionsProps & { children?: ReactNode; }'.
  Type '{ obj: {}[]; }' is not assignable to type 'ConditionsProps'.
    Property 'conditions' is missing in type '{ obj: {}[]; }'.
```

Reply:
Looks like you didn't update where the component is used in routes.tsx line 253

Reply:
Can we be more specific than `obj: any[]`?

For example, in existing code I've seen types like:

```TypeScript  
allkinds: {[s: string]: K8sKind},
// 
match: RouterMatch<{ns: string, plural: string, appName?: string}>;
//
flags: {[name: string]: boolean};
//
maxValues: {[_: string]: number},
totals: {[_: string]: number}
```
To drop a few examples.  If we can capture the shape, even minimally better than `any`, should we do so?

Reply:
ok, I got it working now.  @benjaminapetersen what does the _ signify in `{[_: string]: number}`? -thanks

Reply:
I have no idea. 😆 
Generally speaking, I see complex objects being represented via types by making assertions on some of the key properties.  I think that's the main takeaway.   Dabbling with this myself as well.

Reply:
Could be a Q for @alecmerdler, if he has any advice on the best way to make the most of typescript in these cases.

Reply:
Pondering the difference between types like these:

https://github.com/openshift/console/blob/master/frontend/public/components/cloud-services/index.tsx#L85

And then types that are `props` for Components.  Not sure if there is a dividing line between how specific we should be with `prop` types vs other types.  

Thinking out loud.